src/test/unit_tests/test_namespace_fs.js

```diff
@@ -1764,8 +1764,111 @@ mocha.describe('namespace_fs copy object', function() {
+            const xattr = { key: 'key1', key2: 'value1' };
+
+            mocha.it('Upload object with tagging', async function() {
+                const params = {
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                    xattr,
+                    source_stream: buffer_utils.buffer_to_read_stream(data),
+                    tagging: tagging_1,
+                };
+                const upload_res = await ns_tmp_tagging.upload_object(params, dummy_object_sdk);
+                console.log('upload_object (tag) response', inspect(upload_res));
+                const tag_res = await ns_tmp_tagging.get_object_tagging({
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                }, dummy_object_sdk);
+                console.log('get_object_tagging response', inspect(tag_res));
+                await validate_tagging(tag_res, tagging_1);
+            });
+
+            mocha.it('Upload object with tagging, update object and tag', async function() {
+                const tagging_2 = [{ key: 'tag2', value: 'val2' }];
+                const params = {
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                    xattr,
+                    source_stream: buffer_utils.buffer_to_read_stream(data),
+                    tagging: tagging_2,
+                };
+                const upload_res = await ns_tmp_tagging.upload_object(params, dummy_object_sdk);
+                console.log('upload_object (tag) response', inspect(upload_res));
+                const tag_res = await ns_tmp_tagging.get_object_tagging({
+                    bucket: upload_bkt_tagging,
+                    key: upload_key,
+                }, dummy_object_sdk);
+                console.log('get_object_tagging response', inspect(tag_res));
+                await validate_tagging(tag_res, tagging_2);
+            });
+
+            mocha.it('Upload folder object with tagging', async function() {
+                const params = {
+                    bucket: upload_bkt_tagging,
+                    key: upload_folder_key,
+                    xattr,
+                    source_stream: buffer_utils.buffer_to_read_stream(data),
+                    tagging: tagging_1,
+                };
+                const upload_res = await ns_tmp_tagging.upload_object(params, dummy_object_sdk);
+                console.log('upload_object (tag) response', inspect(upload_res));
+                const tag_res = await ns_tmp_tagging.get_object_tagging({
+                    bucket: upload_bkt_tagging,
+                    key: upload_folder_key,
+                }, dummy_object_sdk);
+                console.log('get_object_tagging response', inspect(tag_res));
+                await validate_tagging(tag_res, tagging_1);
+            });
+
+            mocha.it('Upload folder object with tagging, get key wthout slash', async function() {
```

Comment:
Adding a minor suggestion if you want
```suggestion
            mocha.it('Upload folder object with tagging, get object tagging key without slash', async function() {
```